external/storm-kafka/src/jvm/storm/kafka/PartitionManager.java

```diff
@@ -137,8 +137,14 @@ public EmitState next(SpoutOutputCollector collector) {
             }
             Iterable<List<Object>> tups = KafkaUtils.generateTuples(_spoutConfig, toEmit.msg);
             if (tups != null) {
-                for (List<Object> tup : tups) {
-                    collector.emit(tup, new KafkaMessageId(_partition, toEmit.offset));
+		if(_spoutConfig.topicAsStreamId) {
+	            for (List<Object> tup : tups) {
```

Comment:
can you if inside the loop instead of having two loops. 
for(List<Object> tup: tups) {
        if (_spoutconfig.topicAsStreamId) 
             collector.emit(_spoutConfig.topic, tup, ...);
       else
            collector.emit(tup, new KafkaMessageId..);
}
since you are here please rename tup -> tuple and tups -> tuples
}